xarray/test/test_dataarray.py

```diff
@@ -1407,6 +1407,42 @@ def test_resample_first(self):
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_resample_first_keep_attrs(self):
+        times = pd.date_range('2000-01-01', freq='6H', periods=10)
+        array = DataArray(np.arange(10), [('time', times)])
+        array.attrs['meta'] = 'data'
+
+        resampled_array = array.resample('1D', dim='time', how='first', keep_attrs=True)
```

Comment:
There are a couple reasons not to - it may be unclear which "how" caused the test to fail (though that can usually be fixed with good error messages), and it will often be unclear whether multiple "how"s would cause the test to fail. For "first" and "mean" in particular, different bugs may cause different ones (or both) to fail.

I'll consider this though as I clean the tests up.